### PR TITLE
added "exiting" player status

### PR DESCRIPTION
**Pull request by jolange**
When leaving the main loop, the player status is set to "exiting" and
the status program is spawned just with the arguments "status exiting".

Partially implements #640.

### CODE REVIEW

**Comment by mahkoh** [comment]
I think I've already rejected such a PR in the past.

**Comment by mahkoh** [comment]
#325

**Comment by jolange** [comment]
Thanks for digging this up, I have not seen it before. I will look into the dbus/mpris API, sounds nice. By the way: this is not documented anywhere, is it?

Nevertheless, since cmus provides the interface via the status program, I think cmus' last sign of life should somehow indicate that it is terminating. Why do you think an exiting state is not a good idea?

**Comment by mahkoh** [comment]
FWIW, you can simulate this by using a small wrapper

```bash
#!/bin/sh
cmus
status_program status exiting
```

**Comment by mahkoh** [comment]
Something like this would probably get accepted:

```c
static void spawn_status_program_inner(const char *status_text, struct track_info ti)
{
	if (status_display_program == NULL || status_display_program[0] == 0)
		return;

	char *argv[32];
	int i = 0;

	argv[i++] = xstrdup(status_display_program);
	argv[i++] = xstrdup("status");
	argv[i++] = xstrdup(status_text);
	if (ti)
		expand_status_program_ti_args(ti, argv, &i);
	argv[i++] = NULL;

	if (spawn(argv, NULL, 0) == -1)
		error_msg("couldn't run `%s': %s", status_display_program, strerror(errno));

	for (i = 0; argv[i]; i++)
		free(argv[i]);
}
```

**Comment by jolange** [comment]
@mahkoh, I guess you want to call the 'inner' function after the main loop like this?

File: ui_curses.c [hunk 1]
```diff
@@ -1794,40 +1794,38 @@ static void clear_error(void)
...
 
-	i = 0;
 	argv[i++] = xstrdup(status_display_program);
 
 	argv[i++] = xstrdup("status");
-	argv[i++] = xstrdup(player_status_names[status]);
-	if (player_info.ti) {
+	argv[i++] = xstrdup(status_text);
+
+	if (ti) {
```
Comment (nefthy) [inline ui_curses.c]: please extract this to a seperate function to keep the line count per function resonable.

**Comment by jolange** [comment]
Done. Let me know if you are satisfied with this, then I can rebase it.

(BTW nice to see that you try to revive the development here!)

**Comment by nefthy** [comment]
LGTM, we can merge after the rebase. Thanks.

**Comment by jolange** [comment]
Ok, done!

**Pull request merged**